ipsource.go

```diff
@@ -238,6 +241,70 @@ func (u UPnP) GetIPs(ctx context.Context, _ IPVersions) ([]net.IP, error) {
 	return []net.IP{ip}, nil
 }
 
+// CMD is an IP source that looks up the public IP addresses by
+// executing a script or command from your filesystem.
+//
+// The command must return the IP addresses comma spreaded in plain text.
+type CMD struct {
+	Command string `json:"cmd,omitempty"`
+
+	logger *zap.Logger
+}
+
+// CaddyModule returns the Caddy module information.
+func (CMD) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "dynamic_dns.ip_sources.simple_http",
+		New: func() caddy.Module { return new(CMD) },
+	}
+}
+
+func (c *CMD) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	var (
+		unused string
+		cmd    string
+	)
+	if !d.AllArgs(&unused, &cmd) {
+		return d.ArgErr()
+	}
+	c.Command = cmd
+	return nil
+}
+
+// Provision sets up the module.
+func (c *CMD) Provision(ctx caddy.Context) error {
+	c.logger = ctx.Logger(c)
+	return nil
+}
+
+// GetIPs gets the public addresses of this machine.
+func (c CMD) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP, error) {
+	out := []net.IP{}
+	command := c.Command
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command("bash", "-c", command)
```

Comment:
Bash is not necessarily available on the system. The command should be called directly instead.

Consider how https://github.com/mholt/caddy-events-exec/blob/master/module.go is implemented.